### ISSUE TITLE
Provide all retrieved documents as source in the citation section

### ISSUE THREAD

**Issue by llmon22** [Azure-Samples/azure-search-openai-demo#1137]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [x] feature request
- [x] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> Ask a question/Chat -> Ask question -> view citations

### Any log messages given by the failure
>

### Expected/desired behavior
> I understand that the code for the citations is in AnswerParser.tsx (?). How can I change the code - or any other code If my assumption is wrong - so that all retrieved documents are provided in the citation section?
Also it would be great to be able to split the citation-section into 2 parts: one as is (sources that contain facts used in the answer) & a second part like "related topics/also see" (the rest of the retrieved documents)

### OS and Version?
> Windows 10

### azd version?
> 1.4.2

### Versions
>

### Mention any other details that might be useful

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by llmon22** [comment]
Update:
I successfully changed the code in AnswerParser.tsx by reusing the same logic done for the "fragments" for a new variable "otherFragments". The whole new code looks like this: 

`import { renderToStaticMarkup } from "react-dom/server";
import { getCitationFilePath } from "../../api";

type HtmlParsedAnswer = {
    answerHtml: string;
    citations: string[];
    otherSources: string[];
};

export function parseAnswerToHtml(answer: string, isStreaming: boolean, onCitationClicked: (citationFilePath: string) => void): HtmlParsedAnswer {
    const citations: string[] = [];
    const otherSources: string[] = [];

    // trim any whitespace from the end of the answer after removing follow-up questions
    let parsedAnswer = answer.trim();

    // Omit a citation that is still being typed during streaming
    if (isStreaming) {
        let lastIndex = parsedAnswer.length;
        for (let i = parsedAnswer.length - 1; i >= 0; i--) {
            if (parsedAnswer[i] === "]") {
                break;
            } else if (parsedAnswer[i] === "[") {
                lastIndex = i;
                break;
            }
        }
        const truncatedAnswer = parsedAnswer.substring(0, lastIndex);
        parsedAnswer = truncatedAnswer;
    }

    const parts = parsedAnswer.split(/\[([^\]]+)\]/g);
    const otherParts = parsedAnswer.split(/\|([^\|]+)\|/g);

    const fragments: string[] = parts.map((part, index) => {
        if (index % 2 === 0) {
            return part;
        } else {
            let citationIndex: number;
            if (citations.indexOf(part) !== -1) {
                citationIndex = citations.indexOf(part) + 1;
            } else {
                citations.push(part);
                citationIndex = citations.length;
            }

            const path = getCitationFilePath(part);

            return renderToStaticMarkup(
                <a className="supContainer" title={part} onClick={() => onCitationClicked(path)}>
                    <sup>{citationIndex}</sup>
                </a>
            );
        }
    });

    const otherFragments: string[] = otherParts.map((part, index) => {
        if (index % 2 === 0) {
            return part;
        } else {
            let citationIndex: number;
            if (otherSources.indexOf(part) !== -1) {
                citationIndex = otherSources.indexOf(part) + 1;
            } else {
                otherSources.push(part);
                citationIndex = otherSources.length;
            }

            const path = getCitationFilePath(part);

            return renderToStaticMarkup(
                <a className="supContainer" title={part} onClick={() => onCitationClicked(path)}>
                    <sup>{citationIndex}</sup>
                </a>
            );
        }
    });

    const allFragments: string[] = fragments.concat(otherFragments);

    return {
        answerHtml: allFragments.join(""),
        citations,
        otherSources
    };
}
`
It can be rendered in the Answer.tsx accordingly: 

`{!!parsedAnswer.otherSources.length && (
                <Stack.Item>
                    <Stack horizontal wrap tokens={{ childrenGap: 5 }}>
                        <span className={styles.citationLearnMore}>Other Sources:</span>
                        {parsedAnswer.otherSources.map((x, i) => {
                            const path = getCitationFilePath(x);
                            return (
                                <a key={i} className={styles.citation} title={x} onClick={() => onCitationClicked(path)}>
                                    {`${++i}. ${x}`}
                                </a>
                            );
                        })}
                    </Stack>
                </Stack.Item>
            )}`

I changed the sample conversation in retrievethenread.py to this (bold text is new): 
` # shots/sample conversation
    question = """
'What is the deductible for the employee plan for a visit to Overlake in Bellevue?'

Sources:
info1.txt: deductibles depend on whether you are in-network or out-of-network. In-network deductibles are $500 for employee and $1000 for family. Out-of-network deductibles are $1000 for employee and $2000 for family.
info2.pdf: Overlake is in-network for the employee plan.
info3.pdf: Overlake is the name of the area that includes a park and ride near Bellevue.
info4.pdf: In-network institutions include Overlake, Swedish and others in the region
"""
    answer = "In-network deductibles are $500 for employee and $1000 for family [info1.txt] and Overlake is in-network for the employee plan [info2.pdf][info4.pdf]. **Other sources: |info3.pdf|**"`

So basically just a rework of existing code. It does work sometimes. The issue now is, that I didn't find a prompt context, that is stable enough to reliably return the "other sources". I added something like this: "For those sources that you did not include, because no fact in your response was based on that source: Include the source name - ONLY the source name - after your response under \"Other sources:\"". 

If anyone is interested in this feature, I am happy about your opinion on this approach and/or better working prompt contexts to ensure a stable behaviour.
